app/celery/letters_pdf_tasks.py

```diff
@@ -219,10 +220,18 @@ def process_virus_scan_passed(self, filename):
     if not sanitise_response:
         new_pdf = None
     else:
+        sanitise_response = sanitise_response.json()
         try:
-            new_pdf = base64.b64decode(sanitise_response.json()["file"].encode())
+            new_pdf = base64.b64decode(sanitise_response["file"].encode())
         except JSONDecodeError:
             new_pdf = sanitise_response.content
+
+        if "redaction_failed_message" in sanitise_response and sanitise_response["redaction_failed_message"]:
```

Comment:
you can make this marginally cleaner by a) using `.get` and b) using a tmp variable. eg:

```python
msg = sanitise_response.get('redaction_failed_message')
if msg:
    logger(msg)
```

Reply:
also can we get a unit test for this code flow? 🙏